datafusion/core/src/physical_optimizer/sort_pushdown.rs

```diff
@@ -107,6 +107,12 @@ fn pushdown_sorts_helper(
             // Make sure this `SortExec` satisfies parent requirements:
             let sort_reqs = requirements.data.ordering_requirement.unwrap_or_default();
             // It's possible current plan (`SortExec`) has a fetch value.
+            // And if both of them have fetch values, we should use the minimum one.
```

Comment:
> It's difficult for me to construct such ut, I didn't construct a test that will go to the condition

I don't think it is possible today to hit this code (as the `requirements` will contain the sort correctly)

Thus I think it would be fine to merge this as is in my opinion, or perhaps you could add a check that the requirement_fetch is <= the sort_fetch if present